test/parallel/test-child-process-spawn-typeerror.js

```diff
@@ -162,17 +168,72 @@ assert.doesNotThrow(function() { execFile(cmd, c, n); });
 // string is invalid in arg position (this may seem strange, but is
 // consistent across node API, cf. `net.createServer('not options', 'not
 // callback')`
-assert.throws(function() { execFile(cmd, s, o, c); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, a, s, c); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, a, o, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, a, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, o, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, u, u, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, n, n, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, a, u, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, a, n, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, u, o, s); }, invalidArgValueError);
-assert.throws(function() { execFile(cmd, n, o, s); }, invalidArgValueError);
+invalidSignatureError(
+  () => execFile(cmd, s, o, c),
+  'Invoking child_process.execFile() with invalid arguments: ' +
```

Comment:
We could even put the valid signature here like this:

```
Invoking child_process.execFile(file[, args][, options][, callback]) with invalid arguments: ('${cmd}', 'string', {}, [Function: c])
```

I was just worrying that:

1. The whole line could be too long to read
2. I am not sure if the optional argument syntax themselves are user-friendly or not, especially to people who are unfamiliar with them. Also, some overloaded APIs have rather cumbersome signatures that are hard to express with this optional argument syntax.

Reply:
I would definitely want to get the optional argument syntax in it. Otherwise I fear it is not much help over the situation before.

In the `assert` module errors are now actually spread across multiple lines and I think that is also fine to do here. That way it is possible to have a very nice looking error message like e.g.:

```
`child_process.execFile(file[, args][, options][, callback])` got invoked with an unknown argument combination:

`child_process.execFile('${cmd}', 'string', {}, [Function: c])`
```

Or

```
`child_process.execFile('${cmd}', 'string', {}, [Function: c])` is a unknown argument combination for the signature:

`child_process.execFile(file[, args][, options][, callback])`
```

Or

```
`child_process.execFile()` got called with the following arguments: ('${cmd}', 'string', {}, [Function: c]).
This is a unknown combination for the signature:
`child_process.execFile(file[, args][, options][, callback])`
```